backend/czi_hosted/data_common/dataset_metadata.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+from http import HTTPStatus
+
+import requests
+from backend.common.utils.utils import path_join
+from backend.common.errors import DatasetAccessError
+import backend.czi_hosted.common.rest as common_rest
+from backend.czi_hosted.common.config.app_config import AppConfig
+from backend.czi_hosted.common.config.server_config import ServerConfig
+
+
+def request_dataset_metadata_from_data_portal(data_portal_api_base: str, explorer_url: str):
+    """
+    Check the data portal metadata api for datasets stored under the given url_path
+    If present return dataset metadata object else return None
+    """
+    headers = {"Content-Type": "application/json", "Accept": "application/json"}
+    try:
+        response = requests.get(
+            url=f"{data_portal_api_base}/datasets/meta?url={explorer_url}",
+            headers=headers
+        )
+        if response.status_code == 200:
+            dataset_identifiers = json.loads(response.body)
+            return dataset_identifiers
+        else:
+            return None
+    except Exception:
+        return None
+
+def extrapolate_dataset_location_from_config(server_config: ServerConfig, dataset_explorer_location: str):
+    """
+    Use the dataset_explorer_location and the server config to determine where the dataset is stored
+    """
+    # TODO @mdunitz remove after fork, update config to remove single_dataset option, the multiroot lookup will need to
+    #  remain while we support covid 19 cell atlas
```

Comment:
I suspect the real requirement is not covid19cellatlas support, but need for stand-alone data collections that the Data Directory (aka Portal) does not manage.  Suggest getting this issue onto PM roadmap questions for a decision.

Reply:
I will review with @ambrosejcarr. 

Thanks for reaching out @MDunitz.